gnocchi/storage/__init__.py

```diff
@@ -198,7 +200,9 @@ def get_measures(self, metric, from_timestamp=None, to_timestamp=None,
         :param resample: The granularity to resample to.
         """
         if aggregation not in metric.archive_policy.aggregation_methods:
-            raise AggregationDoesNotExist(metric, aggregation)
+            if granularity is None:
+                granularity = metric.archive_policy.definition[0].granularity
+            raise AggregationDoesNotExist(metric, aggregation, granularity)
```

Comment:
ok. but it literally validates this right here and knows exactly if it's the granularity or the aggregate that is wrong... imo, if a storage driver does not find the actual corresponding file/object because it's yet to receive data, it should not return an error but return an empty timeseries becaues that's exactly what it is.

that said, i imagine this is because you are trying to address #548? is this the only way to handle it?

Reply:
No, I'm actually targeting #175. For that I need to remove archive policy usage from the storage driver and move the actual detection of invalid requests to the API layer.

Reply:
i see. so how do you envision 175 working? there is no validation and the user can submit any combination of granularity+aggregate? because if there's validation at some point prior to this, i'm thinking it's still better to just return empty ts rather than error.

Reply:
>  the user can submit any combination of granularity+aggregate?

Exactly.

> i'm thinking it's still better to just return empty ts rather than error.

Actually, it's better to raise an error in storage and leave the layer above (e.g. REST API) decides what behaviour it wants to implement

That being said, this PR do not change the current behaviour of the API or even of storage – it merges 2 different exception into one. I feel like we're talking about something that is not there (yet).

Reply:
that's fair. i guess as long as we eventually move this validation stuff from storage then.